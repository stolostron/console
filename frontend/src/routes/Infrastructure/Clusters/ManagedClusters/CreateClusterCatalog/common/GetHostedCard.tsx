/* Copyright Contributors to the Open Cluster Management project */
import { CheckIcon, ExternalLinkAltIcon } from '@patternfly/react-icons'
import {
  CatalogCardItemType,
  CatalogColor,
  getPatternflyColor,
  ICatalogCard,
  PatternFlyColor,
} from '@stolostron/react-data-view'
import { TFunction } from 'react-i18next'
import { DOC_LINKS } from '../../../../../../lib/doc-util'

function GetHostedCard(onNext: () => void, t: TFunction, isHypershiftEnabled: boolean, isCLI: boolean): ICatalogCard {
  return {
    id: 'hosted',
    title: t('Hosted'),
    items: [
      {
        type: CatalogCardItemType.Description,
        description: t(
          'Run an OpenShift cluster where the control plane is decoupled from the data plane, and is treated like a multi-tenant workload on a hosting service cluster. The data plane is on a separate network domain that allows segmentation between management and workload traffic.'
        ),
      },
      {
        type: CatalogCardItemType.List,
        title: t(''),
        icon: <CheckIcon color={getPatternflyColor(PatternFlyColor.Green)} />,
        items: [
          {
            text: t('Reduces costs by efficiently reusing an OpenShift cluster to host multiple control planes.'),
          },
          { text: t('Quickly provisions clusters.') },
        ],
      },
    ],
    onClick: isHypershiftEnabled ? onNext : undefined,
    alertTitle: isHypershiftEnabled
      ? undefined
      : t('Hosted control plane operator must be enabled in order to continue'),
    alertVariant: 'info',
    alertContent: (
      <a href={DOC_LINKS.HOSTED_ENABLE_FEATURE_AWS} target="_blank" rel="noopener noreferrer">
        {t('View documentation')} <ExternalLinkAltIcon />
      </a>
    ),
    badgeList: isCLI
      ? [
          {
            badge: t('CLI-based'),
            badgeColor: CatalogColor.purple,
          },
        ]
      : [],
  }
}
export default GetHostedCard
