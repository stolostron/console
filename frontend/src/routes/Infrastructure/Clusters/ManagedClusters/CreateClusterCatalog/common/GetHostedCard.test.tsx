/* Copyright Contributors to the Open Cluster Management project */

import i18next from 'i18next'
import GetHostedCard from './GetHostedCard'

const t = i18next.t.bind(i18next)

describe('GetHostedCard', () => {
  test('Return card data - Hypershift enabled, CLI based', async () => {
    const hostedCard = GetHostedCard(() => {}, t, true, true)
    expect(hostedCard).toMatchSnapshot()
  })

  test('Return card data - Hypershift disabled, non-CLI based', async () => {
    const hostedCard = GetHostedCard(() => {}, t, false, false)
    expect(hostedCard).toMatchSnapshot()
  })
})
