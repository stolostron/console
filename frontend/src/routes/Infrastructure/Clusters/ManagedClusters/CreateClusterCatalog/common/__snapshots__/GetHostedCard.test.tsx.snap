// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetHostedCard Return card data - Hypershift enabled 1`] = `
Object {
  "alertContent": <a
    href="https://docs.redhat.com/en/documentation/red_hat_advanced_cluster_management_for_kubernetes/2.13/html-single/clusters/index#enable-or-disable-hosted-control-planes"
    rel="noopener noreferrer"
    target="_blank"
  >
    View documentation
     
    <ExternalLinkAltIcon
      color="currentColor"
      noVerticalAlign={false}
      size="sm"
    />
  </a>,
  "alertTitle": undefined,
  "alertVariant": "info",
  "id": "hosted",
  "items": Array [
    Object {
      "description": "Run an OpenShift cluster where the control plane is decoupled from the data plane, and is treated like a multi-tenant workload on a hosting service cluster. The data plane is on a separate network domain that allows segmentation between management and workload traffic.",
      "type": "description",
    },
    Object {
      "icon": <CheckIcon
        color="var(--pf-global--success-color--100)"
        noVerticalAlign={false}
        size="sm"
      />,
      "items": Array [
        Object {
          "text": "Reduces costs by efficiently reusing an OpenShift cluster to host multiple control planes.",
        },
        Object {
          "text": "Quickly provisions clusters.",
        },
      ],
      "title": "",
      "type": "list",
    },
  ],
  "onClick": [Function],
  "title": "Hosted",
}
`;
